src/app/services/registry.service.ts

```diff
@@ -483,7 +483,13 @@ export class PluginRegistryBaseService {
 
     public async getByApiLink<T>(link: ApiLink, searchParams: URLSearchParams | null = null, ignoreCache: boolean | "ignore-embedded" = false): Promise<ApiResponse<T> | null> {
         const url = new URL(link.href)
-        searchParams?.forEach((value, key) => url.searchParams.append(key, value));
+        searchParams?.forEach((value, key) => {
+            if (url.searchParams.has(key)) {
+                url.searchParams.set(key, value);
+            } else {
+                url.searchParams.append(key, value);
+            }
+        });
```

Comment:
In the course of this pull request, I adjusted the `getByApiLink` method to replace existing parameters. The reason for this change was that setting the `item-count` URL parameter via the `searchParmas` for [`getByRel(['ui-template', 'collection'], searchParams)`](https://github.com/UST-QuAntiL/qhana-ui/blob/f6b830ca64462f9698eb7d269ecae19dedfe1e39/src/app/services/registry.service.ts#L459) didn't work. The `ApiLink` (given to `getByApiLink`) already contains the default parameter `item-count=25` and appending the parameter a second time doesn't work (Angular seems to use only the first occurrence).
As, currently, loading the templates is handled by a `qhana-growing-list` this change could be reverted. Does it make sense to revert this change?

Reply:
No, keep your version. It is better if the extra url params actually are effective. However, in the future try to keep changes to such core/common parts in separate commits. This makes it easy to recognize these changes, and revert them later if necessary. It also allows you to write why that change was required in the first place into the commit message (not in the summary line, but the extended body). I use this to document smaller design decisions that are only ever relevant if one takes a look at the code.